### ISSUE TITLE
Improve documentation on Configuration page

### ISSUE THREAD

**Issue by Samreay** [apache/iceberg-python#1290]
### Feature Request / Improvement

Hi team! We're currently adopting Iceberg format but struggling to configure everything. For example, we'd like to change the parquet compression code from its default to snappy.

I can see that this is an option in the configuration:

![image](https://github.com/user-attachments/assets/0dd40a32-4c8b-4894-a2fb-de9e91bc735a)

But _how_ to actually use the configuration options and apply them seems to be missing from the page. Is there a `pyiceberg.configure` method, are these options that should be passed into `.append`, should this be set as an environment variable with some form of translation from `write.parquet.compression-codec` to `PYICEBERG_WRITE_PARQUET_COMPRESSION-CODEC`?

I'm not sure!

I'd gladly open a PR on this (once I know how you're supposed to do it) to add that detail, and also fix the incorrect description in the `write.parquet.page-row-limit` and `row-group-limit` rows.

**Comment by Fokko** [comment]
> Later on in the page in the Catalogs section, there are some ways detailed as to how to configure catalogs, but I'm not sure if this applies to writer configuration as well (why would it not be at the top if it applied to everything).

Yes, this is indeed the case. Sorry for the confusion there, and thanks for raising a PR. These PRs are very helpful to the community, since the folks that already use PyIceberg know what to look for, and skip parts of the documentation.

**Comment by Samreay** [comment]
@Fokko I've updated the lint in the PR. Also, what are your thoughts on enabling Github discussions? I have a few questions about best practises and similar but I don't want to clutter Issues with things that aren't actual issues.

**Comment by Fokko** [comment]
Enabling Github Discussions has been brought up once or twice, but it hasn't been yet decided. Mainly because there are [several places to discuss things](https://iceberg.apache.org/community/#issues), for example, Slack, GitHub Issues or the ASF mailing list. Feel free to raise Github issues, or start a thread on either the mailinglist or Slack (we have a #python channel).

**Comment by Samreay** [comment]
Thanks @Fokko, I've joined the community slack and made a post to the compaction channel about a different topic, looking forward to getting more engaged with the community. Cheers mate.
